pylivetrader/backend/alpaca.py

```diff
@@ -287,6 +288,27 @@ def orders(self):
             for o in self._api.list_orders('all')
         }
 
+    def all_orders(self, status=None):
+        until = pd.Timestamp.utcnow().isoformat()
+        all_orders = {}
+        batch_size = 500
+        orders = self._api.list_orders(status, batch_size, until=until)
+        while(True):
+            batch_orders = {
+                o.client_order_id: self._order2zp(o)
+                for o in orders
+            }
+            # get the timestamp of the earliest order in the batch
+            # add 1 second to avoid orders being dropped at batch limit
+            until = pd.Timestamp(
+                orders[-1].submitted_at + timedelta(seconds=1)
+            ).isoformat()
+            all_orders = {**all_orders, **batch_orders}
+            orders = self._api.list_orders(status, batch_size, until=until)
+            if len(orders) < batch_size:
```

Comment:
This orders less than batch_size should also be part of `all_orders`, no?

Reply:
They are, this is just for loop control (e.g. breaks out of loop when it got back 150 when it asked for 500, but still does the operation on those 150)

Reply:
You will lose last 100 if the result is 600.